spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/OptionalPayloadTests.java

```diff
@@ -0,0 +1,149 @@
+import org.springframework.amqp.rabbit.junit.RabbitAvailableCondition;
+import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * @author Gary Russell
+ * @since 2.8
+ *
+ */
+@SpringJUnitConfig
+@RabbitAvailable(queues = { "op.1", "op.2" })
+public class OptionalPayloadTests {
+
+	@Test
+	void optionals(@Autowired RabbitTemplate template, @Autowired Listener listener)
+			throws JsonProcessingException, AmqpException, InterruptedException {
+
+		ObjectMapper objectMapper = new ObjectMapper();
+		template.send("op.1", MessageBuilder.withBody(objectMapper.writeValueAsBytes("foo"))
+				.andProperties(MessagePropertiesBuilder.newInstance()
+						.setContentType("application/json")
+						.build())
+				.build());
+		template.send("op.1", MessageBuilder.withBody(objectMapper.writeValueAsBytes(null))
+				.andProperties(MessagePropertiesBuilder.newInstance()
+						.setContentType("application/json")
+						.build())
+				.build());
+		template.send("op.2", MessageBuilder.withBody(objectMapper.writeValueAsBytes("bar"))
+				.andProperties(MessagePropertiesBuilder.newInstance()
+						.setContentType("application/json")
+						.build())
+				.build());
+		template.send("op.2", MessageBuilder.withBody(objectMapper.writeValueAsBytes(null))
+				.andProperties(MessagePropertiesBuilder.newInstance()
+						.setContentType("application/json")
+						.build())
+				.build());
+		assertThat(listener.latch.await(10, TimeUnit.SECONDS)).isTrue();
+		assertThat(listener.deOptionaled).containsExactlyInAnyOrder("foo", null, "bar", "baz");
+	}
+
+	@Configuration
+	@EnableRabbit
+	public static class Config {
+
+		@Bean
+		RabbitTemplate template() {
+			return new RabbitTemplate(rabbitConnectionFactory());
+		}
+
+		@Bean
+		SimpleRabbitListenerContainerFactory rabbitListenerContainerFactory() {
+			SimpleRabbitListenerContainerFactory factory = new SimpleRabbitListenerContainerFactory();
+			factory.setConnectionFactory(rabbitConnectionFactory());
+			factory.setMessageConverter(converter());
+			return factory;
+		}
+
+		@Bean
+		ConnectionFactory rabbitConnectionFactory() {
+			BrokerRunningSupport brokerRunning = RabbitAvailableCondition.getBrokerRunning();
+			CachingConnectionFactory connectionFactory = new CachingConnectionFactory();
+			connectionFactory.setHost(brokerRunning.getHostName());
+			connectionFactory.setPort(brokerRunning.getPort());
+			connectionFactory.setUsername(brokerRunning.getUser());
+			connectionFactory.setPassword(brokerRunning.getPassword());
+			return connectionFactory;
```

Comment:
How about this instead: `return new CachingConnectionFactory(RabbitAvailableCondition.getBrokerRunning().getConnectionFactory());` ?

Reply:
Yeah copy/paste from another test that was exercising these methods.